my-sms/app/controllers/students_controller.rb

```diff
@@ -5,11 +5,14 @@ class StudentsController < ApplicationController
   decorates_assigned :student
 
   before_filter :find_student, only: %i[show edit update destroy]
+  before_render :flash_errors, only: %i[edit create update]
```

Comment:
I'm thinking if you change the `student` to `@student` in the `flash_errors` method you can move this to `before_filter`. Great work on figuring out how to create your own callbacks, but I've not had to use custom callbacks like this before, so I'm a bit suspicious it's required here.

Also another opinion/personal note, I prefer calling the symbols directly rather than use the `%i` macro

```suggestion
  before_filter :flash_errors, only: [:edit, :create, :update]
```
```

Reply:
The reason I had to create before_render is because rails does adds the error messages and the errors themselves to the model at different times. The messages are not added in until render is called. Not sure why, maybe because they want to maintain MVC separation?

Reply:
The errors will be added when the model validations occur. These validations occur when the record is attempted to save to the database (eg. `@student.save`), or when you specifically call the `valid?` method (more info [here](https://guides.rubyonrails.org/active_record_validations.html)). 

Maybe try adding a `return unless @student.valid?` line to your `flash_errors` method.

Reply:
That was a typo sorry 😅. What I meant was the error IDs themselves are added during validation, but corresponding messages for each type of error seem to be only added during render.

Reply:
Hmm OK, let's leave for now.

Maybe in future, we can stick with original of showing errors above (using what you had before) and just indicate a simple message in the flash to indicate there are errors

something like

```ruby
def render_flash
  if @model.invalid?
    flash[:error] = "There are errors with your model"
  end
end
```